### PR TITLE
Consistency Check Failure is Fixed

### PR DESCRIPTION
**Pull request by abukhoy**
BUG: OSError: Consistency check failed: file should be of size {original size} but has size {downloaded size}.
We are sorry for the inconvenience. Please retry with `force_download=True`.
If the issue persists, please let us know by opening an issue on https://github.com/huggingface/huggingface_hub.

--> it is fixed now with this PR.

### CODE REVIEW

File: QEfficient/utils/constants.py [hunk 1]
```diff
@@ -55,3 +55,4 @@ class Constants:
     INPUT_STR = ["My name is"]
     GB = 2**30
     MAX_QPC_LIMIT = 30
+    MAX_RETRIES = 5
```
Comment (quic-rishinr) [inline QEfficient/utils/constants.py:58]: Add a comment mentioning this constant will be used set the maximum number of retry attempts for downloading a model using huggingface_hub snapshot_download

File: QEfficient/utils/_utils.py [hunk 2]
```diff
@@ -67,6 +67,15 @@ def hf_download(
             else:
                 raise e
 
+        except OSError as e:
+            logger.error(f"OSError: {e}")
+            if "Consistency check failed" in str(e):
+                logger.info(
+                    "OSError: Consistency check failed: file should not be incomplete, Resuming the downloading..."
```
Comment (quic-rishinr) [inline QEfficient/utils/_utils.py:74]: rephrase this to something like "Consistency check failed during model download. The file appears to be incomplete. Resuming the download..."

**Review by quic-rishinr — CHANGES_REQUESTED** [review]

File: QEfficient/utils/_utils.py (hunk 2 above)
Comment (abukhoy) [reply QEfficient/utils/_utils.py:74]: will do.

**Comment by abukhoy** [comment]
@quic-rishinr  the changes have been made.

File: QEfficient/utils/_utils.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def hf_download(
     if cache_dir is not None:
         os.makedirs(cache_dir, exist_ok=True)
 
-    max_retries = 5
+    max_retries = Constants.MAX_RETRIES
     retry_count = 0
     while retry_count < max_retries:
```
Comment (ochougul) [inline QEfficient/utils/_utils.py:47]: directly use 
```retry_count<constants.MAX_RETRIES``` instead of redefining on line 45

**Review by ochougul — CHANGES_REQUESTED** [review]

File: QEfficient/utils/_utils.py (hunk 3 above)
Comment (abukhoy) [reply QEfficient/utils/_utils.py:47]: Okay, I will do that.

**Review by quic-amitraj — APPROVED** [review]
LGTM

File: QEfficient/utils/_utils.py [hunk 4]
```diff
@@ -61,12 +60,21 @@ def hf_download(
             retry_count += 1
 
         except HTTPError as e:
-            retry_count = max_retries
+            retry_count = Constants.MAX_RETRIES
```
Comment (quic-amitraj) [inline QEfficient/utils/_utils.py:63]: I think its better to use  `break` statement instead of `retry_count = Constants.MAX_RETRIES`?

Comment (ochougul) [reply QEfficient/utils/_utils.py:63]: We should raise the error here, if we only break, the execution will continue even when the model is not downloaded, and we will get `undefined variable model_path` as error.
Please change this to
```
if e.response.status_code == 401:
    logger.info("You need to pass a valid `--hf_token=...` to download private checkpoints.")
raise e
```

File: QEfficient/utils/_utils.py [hunk 5]
```diff
@@ -42,9 +42,8 @@ def hf_download(
     if cache_dir is not None:
```
Comment (ochougul) [inline QEfficient/utils/_utils.py:42]: can you change the function definition to take max_retries as input and set it's default value to `constants.MAX_RETRIES`

Comment (quic-rishinr) [reply QEfficient/utils/_utils.py:49]: @ochougul this can be a configurable value. so it would be better if we can fetch from a file. If we need to make the max_retries as input we need to provide a way for user to set it.

**Review by quic-rishinr — APPROVED** [review]
